.github/workflows/lint.yml

```diff
@@ -0,0 +1,58 @@
+name: lint
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  luacheck:
+    # We want to run on external PRs, but not on our own internal
+    # PRs as they'll be run by the push to the branch.
+    #
+    # The main trick is described here:
+    # https://github.com/Dart-Code/Dart-Code/pull/2375
+    if: github.event_name == 'push' ||
+      github.event.pull_request.head.repo.full_name != github.repository
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: Install deps
+        run: |
+          sudo apt-get -y -f install lua5.1 luarocks
+          sudo luarocks install luacheck 0.26.1
+      - name: Luacheck
+        run: luacheck --codes --config .luacheckrc .
+
+  checkpatch:
+    if: github.event_name == 'pull_request'
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Install deps
+        uses: actions/checkout@v2
+        with:
+          repository: tarantool/checkpatch
+          path: 'checkpatch'
+
+      - name: Check patch
+        run: |
+          checkpatch/checkpatch.pl --color=always --show-types \
+            --ignore NO_DOC,NO_TEST,NO_CHANGELOG \
```

Comment:
I think it is fine to validate doc and test. Just changelogs are not needed for now.

Reply:
If checkpatch was in CI already, i wouldn't misspelled in the previous commit:
```
Let's disable this test before the fix is commited.
```

It allows to avoid at least this small errors in commit messages. Moreover, I propose not to run it on all pushes, in order not to make CI red when pushing directly to master. In current patch `checkpatch` runs only on pull requests.

However, if you think we don't need it, I won't insist) However, it seems like a pretty useful thing to have on CI.

Reply:
I didn't say we don't need it. On the contrary, I said we need it with doc and test checks enabled too. Берем его весь, игнорим только `NO_CHANGELOG`.

Reply:
My bad) Fixed